source/fundamentals/logging.txt

```diff
@@ -0,0 +1,273 @@
+=======
+Logging
+=======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. _logging:
+
+Overview
+--------
+
+.. What is on this page?
+
+In this guide, we show you how to set up and configure a logger in the
+MongoDB Java driver.
+
+.. What do any new terms mean?
+
+.. Not applicable. We are treating definition of logger as understood.
+
+.. What can you expect to see on this page?
+
+In this guide you can learn how to do the following:
+
+- Set up a logger using the Simple Logging Facade For Java (SLF4J)
+- Configure the log level of your logger
+
+.. Who should read this? Where should I go if this isn't the page I was looking for?
+
+.. Not Applicable. No natural redirect location.
+
+Set Up a Logger
+---------------
+
+To set up a logger for your project, you need to add some dependencies.
+
+The MongoDB Java driver uses the Simple Logging Facade For Java (SLF4J).
+SLF4J allows you to specify your logging framework of choice at deployment time.
+For more information on SLF4J,
+`see the SLF4J documentation <http://www.slf4j.org/>`__.
+
+To set up a logger you must add the ``slf4j-api`` package, a logging framework, and a
+**binding** to your project's dependency list. A binding is a piece of code that
+connects the ``slf4j-api`` package with a logging framework. The two most
+popular logging frameworks used with ``slf4j-api`` are Log4j2 and Logback.
+
+Setting up a logger is optional. When you start your application the MongoDB
+Java driver looks for the ``slf4j-api`` package in your classpath. If the driver
+can't find ``slf4j-api``, the driver logs a single warning with
+``java.util.logging`` and disables all further logging.
+
+Here is the error the driver outputs if it cannot find the ``slf4j-api`` package:
+
+.. code-block:: none
+
+   WARNING: SLF4J not found on the classpath.  Logging is disabled for the 'org.mongodb.driver' component
+
+
+Below is an example showing how to add SLF4J dependencies to your
+project and set up your logger using either Logback or Log4j2. The examples uses Maven.
+
+.. tabs::
+
+   .. tab:: Logback
+      :tabid: logback-binding
+
+      If you add these dependencies to your ``pom.xml`` file, the MongoDB Java
+      driver uses the Logback framework to output your logs.
+
+      .. code-block:: xml
+
+         <!-- SLF4J API -->
+         <dependency>
+           <groupId>org.slf4j</groupId>
+           <artifactId>slf4j-api</artifactId>
+           <version>1.7.30</version>
+         </dependency>
+         <!-- This package contains an SLF4J-Logback binding -->
```

Comment:
The comment here doesn't match comment in the other example. Is there a reason for this?

Reply:
Good question. The reason the comments are different is that the `logback-classic` package includes more than just an SLF4J-Logback binding, where as the `log4j-slf4j-impl` package only contains a logback binding. It felt inaccurate for me to write the comment "SLF4J-Logback Binding" for logback and it felt excessive to write "This package contains an SLF4J-Log4j2 binding" for Log4j2. The two frameworks are different projects, so I think it is okay for the xml comments to be different. Let me know if you disagree and you would like me to make both comments "SLF4J-<framework name>".

Reply:
is there any meaningful difference to the user though? If not, perhaps the importance you are placing on it is for you alone?

Reply:
Thanks for the note. Assuming a user copies and pastes these dependencies into their code, I think having the comment accurately reflect the contents of the package will meaningfully benefit users. The logback classic has more than just an `SLF4J-Logback` binding, so if a user ever wants to clean up their `pom.xml` file and finds these dependencies there without context, I think "contains an `SLF4J-Logback binding` more accurately expresses that "this was added here because of `SLF4J`, but this contains things that may not be expressly related to SLF4J, becareful about removing it".

Reply:
Let me know if you disagree, it you want me to just make the change I can. I do actually feel the different comment is beneficial, but my instincts could just be incorrect, in which case I am totally open to making comments identical.